graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java

```diff
@@ -200,8 +205,9 @@ private DirectIndexQueryResult performAsyncDirectIndexQuery(SearchParams searchP
                 }
                 response = getAsyncSearchResponse(searchParams, esSearchId).get();
                 if (response ==  null) {
-                    // Return null, if the response is null wil help returning @204 HTTP_NO_CONTENT to the user
-                    return null;
+                    // Rather than null (if the response is null wil help returning @204 HTTP_NO_CONTENT to the user)
```

Comment:
2 things here
1. response == null does not seem like a good check for time out. 204 could be really no result for a valid search with no results. So we need to see what is the correct status code to throw error.
2. Also log the query that got timed out